core/state_transition.go

```diff
@@ -438,167 +432,94 @@ func (st *StateTransition) applyEditValidatorTx(
 		wrapper.Validator.UpdateHeight = blockNum
 	}
 
-	if newRate.Sub(rateAtBeginningOfEpoch).Abs().GT(wrapper.Validator.MaxChangeRate) {
-		return errCommissionRateChangeTooFast
-	}
-
-	if newRate.GT(wrapper.Validator.MaxRate) {
-		return errCommissionRateChangeTooHigh
-	}
-
-	if err := st.state.UpdateStakingInfo(wrapper.Address, wrapper); err != nil {
-		return err
-	}
-	return nil
+	return st.state.UpdateStakingInfo(wrapper.Address, wrapper)
 }
 
+// applyDelegateTx applies the delegate staking transaction.
+// Note that it assumes that the staking message is valid.
 func (st *StateTransition) applyDelegateTx(delegate *staking.Delegate) error {
-	if delegate.Amount.Sign() == -1 {
-		return errNegativeAmount
-	}
-
-	if !st.state.IsValidator(delegate.ValidatorAddress) {
-		return errValidatorNotExist
-	}
 	wrapper := st.state.GetStakingInfo(delegate.ValidatorAddress)
 	if wrapper == nil {
 		return errValidatorNotExist
 	}
-
-	stateDB := st.state
-	delegatorExist := false
-	for i := range wrapper.Delegations {
-		delegation := &wrapper.Delegations[i]
```

Comment:
Use this original way of referencing delegation. Your way gets a copy of the delegation, and it is a source of problems

Reply:
oh alright, changed back to original way